File: robocode-tankroyale-bot-api-java/src/main/java/dev/robocode/tankroyale/botapi/events/BulletFiredEvent.java
package dev.robocode.tankroyale.botapi.events;

import dev.robocode.tankroyale.botapi.BulletState;

/** Event occurring when a bullet has been fired from a bot */
@SuppressWarnings("unused")
public final class BulletFiredEvent extends Event {

  /** Bullet that was fired */
  private final BulletState bullet;

  public BulletFiredEvent(int turnNumber, BulletState bullet) {
    super(turnNumber);
    this.bullet = bullet;
  }

  /** Returns the bullet that was fired */
  public BulletState getBullet() {
    return bullet;
  }
}

File: bot-api/dotnet/api/src/events/BulletFiredEvent.cs
using Newtonsoft.Json;

namespace Robocode.TankRoyale.BotApi.Events;

/// <summary>
/// Event occurring when a bullet has been fired from a bot.
/// </summary>
public sealed class BulletFiredEvent : BotEvent
{
    /// <summary>Bullet that was fired.</summary>
    public BulletState Bullet { get; }

    /// <summary>
    /// Initializes a new instance of the BulletFiredEvent class.
    /// </summary>
    /// <param name="turnNumber">Turn number.</param>
    /// <param name="bullet">Bullet that was fired.</param>
    [JsonConstructor]
    public BulletFiredEvent(int turnNumber, BulletState bullet) : base(turnNumber) => Bullet = bullet;
}